src/proxy/outbound.rs

```diff
@@ -93,25 +100,25 @@ impl Outbound {
                 match socket {
                     Ok((stream, _remote)) => {
                         let mut oc = OutboundConnection {
-                            pi: self.pi.clone(),
+                            pi: pi.clone(),
                             id: TraceParent::new(),
                             pool: pool.clone(),
                         };
                         let span = info_span!("outbound", id=%oc.id);
-                        tokio::spawn(
-                            (async move {
-                                debug!(dur=?start_outbound_instant.elapsed(), id=%oc.id, "outbound spawn START");
-                                // Since this task is spawned, make sure we are guaranteed to terminate
-                                tokio::select! {
-                                        _ = outbound_drain.signaled() => {
-                                            debug!("outbound drain signaled");
-                                        }
-                                        _ = oc.proxy(stream) => {}
+                        let serve_outbound_connection = (async move {
+                            debug!(dur=?start_outbound_instant.elapsed(), id=%oc.id, "outbound spawn START");
+                            // Since this task is spawned, make sure we are guaranteed to terminate
+                            tokio::select! {
+                                _ = outbound_drain.signaled() => {
+                                    debug!("outbound drain signaled");
                                 }
-                                debug!(dur=?start_outbound_instant.elapsed(), id=%oc.id, "outbound spawn DONE");
-                            })
-                            .instrument(span),
-                        );
+                                _ = oc.proxy(stream) => {}
+                            }
+                            debug!(dur=?start_outbound_instant.elapsed(), id=%oc.id, "outbound spawn DONE");
+                        }).instrument(span);
+
+                        assertions::size_between_ref(1000, 1500, &serve_outbound_connection);
```

Comment:
ooh can we selectively enable size assertions via subcomponent logging level toggles?

Reply:
Its not from logging levels, just conditional compiling when we are in test vs release mode

Reply:
I think you can do conditional based on logs, but its a compile time thing so you cannot dynamically change